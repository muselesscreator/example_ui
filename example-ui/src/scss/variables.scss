$panel-light: #bbd;

$panel-dark: #35353a;
$panel-v-dark: #262723;

$disabled-button: #667;

$border-main: gray;
$border-dark: #333;

$green: #94c940;
$darker-green: #352;

$panel-v-light: #d2dbef;
$panel-med-light: #595d66;
$panel-med-dark: #31333d;
$panel-main: #3f444f;
$heading-color: #b3bac8;

$blue: #2185D0;
$green: #21BA45;
$red: #DB2828;
$yellow: #EAAE00;
$orange: #E76B00;
