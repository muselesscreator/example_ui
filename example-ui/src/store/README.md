# Redux Store
## What's in this module?

### index.js (default store import!)
This is the default import point for the redux store in the majority of the project, imported as 
```js
import {actions, selectors, thunkActions, global_selectors} from 'store';
```

### store.js
This file is the source for the above index file, specifically separated out so that there can be a family of global selectors and thunkActions without introducing circular dependencies.
 
### global_selectors
This is the home for selectors that want to access multiple store modules.

### thunkActions
This is the home for thunkActions that want to access multiple store modules, particularly between the argus websocket api and the gui store module.
