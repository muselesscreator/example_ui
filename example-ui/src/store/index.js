/* Main import point for redux store.
 * The base versions of pretty much everything exported from here is actually 
 * defined in './store.js'.  This import trick is so that we can have a set of
 * global selectors with access to the finalized selectors without introducing
 * import loops.
 *
 * The only ting actually changed here is the thunkActions, which merge in a
 * local collection to those imported via connected modules.
 */

import {
  actions,
  reducer,
  selectors,
  thunkActions,
} from './store';

/* This chunk of code is to better enable debugging of the UI in development mode.
 * logs out all actions, thunkActions, selectors, and global_selectors and provides
 * window-level access to them.
 *
 * This is split down a bit as past a certain point, js logging stops noting nested 
 * modules as log-able dicts.
 */
// TODO: Lock this to admin mode?
if (window.name !== 'nodejs') {
  console.log({
    actions,
    selectors,
  });
}

export {
  actions,
  reducer as default,
  selectors,
  thunkActions,
};
