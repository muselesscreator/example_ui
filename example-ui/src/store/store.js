import { connectReducers } from 'js-libs/build/redux';

import * as simple_store from './modules/simple_store';

const {
  thunkActions: _thunkActions, actions, reducer, selectors
} = connectReducers({
  simple_store,
});

export {
  actions,
  reducer,
  selectors
};
export const thunkActions = { ..._thunkActions };
