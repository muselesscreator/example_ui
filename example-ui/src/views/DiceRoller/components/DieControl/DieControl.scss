@import "js-libs/src/scss/colors.scss";
.die-control {
  display: inline-block;
  position: relative;
  width: 200px;
  margin-right: 20px;
  margin-bottom: 50px;
  border: 4px ridge $light_grey_50;
  padding: 20px;
  text-align: center;
  img.die-img {
    width: 100px;
    height: 100px;
    margin-bottom: 20px;
  }
  .last-value {
    height: 40px;
  }
  .ui.button.brw-btn {
    width: 160px;
    height: 60px;
    font-size: 20px !important;
  }

}
