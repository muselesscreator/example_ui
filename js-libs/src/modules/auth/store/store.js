import { action, mkReducer, simpleSelectors } from 'modules/redux';
import { StrictDict } from 'modules/common';

import Auth from '../comms';
import * as thunkActions from './thunkActions';

export const initialState = {
  user: null,
  parseHash: null,
};

export const actions = StrictDict({
  userProfileLoaded: action('USER_PROFILE_LOADED', ['user']),
  handleAuthCallback: action('HANDLE_AUTH_CALLBACK', ['parseHash']),
});

export const reducer = mkReducer(initialState, {
  [actions.userProfileLoaded.type]: (state, { user }) => ({ ...state, user }),
  [actions.handleAuthCallback.type]: (state, { parseHash }) => ({ ...state, parseHash }),
});

export const selectors = {
  ...simpleSelectors(initialState),
  isAuthenticated: ({ user }) => user !== null,
  userName: ({ user }) => user !== null ? Auth.selectors.userName(user) : null,
};

export { thunkActions };
