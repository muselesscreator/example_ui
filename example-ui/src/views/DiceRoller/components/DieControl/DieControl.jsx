/** @module */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { Button, Component } from 'components';
import { actions, selectors } from 'store';

import {
  diceSides,
  diceURIs,
} from 'variables/dice';

import './DieControl.scss';

/**
 * Simple Dice-roll control component.
 * Provides a button for rolling the die, a visual display of the die type,
 * and a textual display of the most recent roll of this die.
 *
 * @param {string} type - die type ({ diceTypes } from 'variables/dice')
 */
export class DieControl extends Component {
  constructor(props) {
    super(props);
    this.state = {
      lastValue: undefined,
    };
    this.roll = this.roll.bind(this);
  }

  componentDidUpdate(prevProps, prevState) {
    // reset lastValue when rolls are cleared.
    if (this.props.rolls.length === 0 && this.state.lastValue) {
      this.setState({ lastValue: undefined });
    }
  }

  /**
   * Generate a new value based on the die type, update lastValue,
   * and call addRoll prop with roll string.
   */
  roll() {
    const sides = diceSides[this.props.type];
    const value = Math.floor(Math.random() * sides + 1);
    this.setState({ lastValue: value });
    this.props.addRoll(`You rolled a(n) ${value} on a ${this.props.type}`);
  }

  render() {
    const { type } = this.props;
    return (
      <div className="die-control">
        <img
          className="die-img"
          src={diceURIs[type]}
          alt="dice"
        />
        <br />
        <div className="last-value">
          { this.state.lastValue && `Last Value: ${this.state.lastValue}`}
        </div>
        <Button className="roll-btn" onClick={this.roll}>
          Roll 1{type}
        </Button>
      </div>
    );
  }
}

DieControl.propTypes = {
  type: PropTypes.string,
  // redux
  addRoll: PropTypes.func,
  rolls: PropTypes.arrayOf(PropTypes.string),
};

export const mapStateToProps = (state) => ({
  rolls: selectors.simple_store.rolls(state),
});

export const mapDispatchToProps = {
  addRoll: actions.simple_store.addRoll,
};

export default connect(mapStateToProps, mapDispatchToProps)(DieControl);
