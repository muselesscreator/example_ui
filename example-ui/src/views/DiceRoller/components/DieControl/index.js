import DieControl from './DieControl';

export default DieControl;
