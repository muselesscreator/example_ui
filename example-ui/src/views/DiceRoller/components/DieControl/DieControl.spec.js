import { components } from 'js-libs/build/redux-test';

import * as module from './DieControl';

import * as variables from 'variables/dice';

import { selectors, actions } from 'store';

jest.mock('store', () => ({
  selectors: {},
  actions: {
    simple_store: {
      addRoll: jest.fn().mockName('addRoll'),
    },
  },
}));

describe('DieControl component', () => {
  describe('Component', () => {
    let mocks = [];
    const mockMethod = (path, impl) => {
      let mock = jest.spyOn(module.DieControl.prototype, path)
        .mockImplementation(impl !== undefined ? impl : () => ({}))
        .mockName(path);
      mocks.push(mock);
    }
    const props = {
      type: variables.diceTypes.d8,
      addRoll: jest.fn().mockName('addRoll'),
      rolls: ['some', 'rolls'],
    };

    afterEach(() => {
      mocks.forEach(mock => mock.mockRestore());
    });

    describe('snapshots', () => {
      beforeEach(() => {
        mockMethod('roll');
      });
      components.simpleTest('smoke test', module.DieControl, props);
    });

    describe('behavior', () => {
      let el, inst;
      beforeEach(() => {
        el = components.render(module.DieControl, props);
        inst = el.instance();
      });
      test('initialState - lastValue should default to undefined', () => {
        expect(el.state().lastValue).toEqual(undefined);
      });
      test('roll', () => {
        jest.spyOn(global.Math, 'random').mockReturnValue(0.999);
        inst.roll();
        expect(el.state().lastValue).toEqual(8);
        expect(props.addRoll).toHaveBeenCalledWith("You rolled a(n) 8 on a d8");
      });
      test('clear rolls on update', () => {
        el.setState({ lastValue: 12 });
        el.setProps({ rolls: [] });
        expect(el.state().lastValue).toEqual(undefined);
      });
    });
  });
  describe('mapSateToProps', () => {
    components.testStateToProps(
      module.mapStateToProps,
      selectors,
      {
        rolls: 'simple_store.rolls',
      }
    );
  });
  describe('mapDispatchToProps', () => {
    const { mapDispatchToProps: props } = module;
    test('addRoll - linked to simple_store action', () => {
      expect(props.addRoll).toEqual(actions.simple_store.addRoll);
    });
  });

});

