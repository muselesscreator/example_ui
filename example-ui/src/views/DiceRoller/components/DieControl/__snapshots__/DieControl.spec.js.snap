// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DieControl component Component snapshots :smoke test 1`] = `
<div
  className="die-control"
>
  <img
    alt="dice"
    className="die-img"
    src="https://s3.amazonaws.com/files.thegamecrafter.com/c123717767de732ae400c7cc0e193581ee444987"
  />
  <br />
  <div
    className="last-value"
  />
  <Button
    className="roll-btn"
    onClick={[Function]}
  >
    Roll 1
    d8
  </Button>
</div>
`;
